modules/vpc-endpoints/main.tf

```diff
@@ -0,0 +1,42 @@
+locals {
+  enabled = module.this.enabled && (length(var.gateway_vpc_endpoints) > 0 || length(var.interface_vpc_endpoints) > 0)
+}
+
+data "aws_vpc_endpoint_service" "gateway_endpoint_service" {
+  for_each     = local.enabled ? var.gateway_vpc_endpoints : {}
+  service      = var.gateway_vpc_endpoints[each.key].name
+  service_type = "Gateway"
+}
+
+data "aws_vpc_endpoint_service" "interface_endpoint_service" {
+  for_each     = local.enabled ? var.interface_vpc_endpoints : {}
+  service      = var.interface_vpc_endpoints[each.key].name
+  service_type = "Interface"
+}
+
+resource "aws_vpc_endpoint" "gateway_endpoint" {
+  for_each          = local.enabled ? data.aws_vpc_endpoint_service.gateway_endpoint_service : set()
+  service_name      = data.aws_vpc_endpoint_service.gateway_endpoint_service[each.key].service_name
+  policy            = var.gateway_vpc_endpoints[each.key].policy
+  vpc_endpoint_type = data.aws_vpc_endpoint_service.gateway_endpoint_service[each.key].service_type
+  vpc_id            = var.vpc_id
+
+  tags = merge(
+    module.this.tags,
+  { Name = "${module.this.id}-${each.key}" })
```

Comment:
@korenyoni 

we usually don't manually override tags, we use a separate label module and add attributes to it

```
module "gateway_endpoint_label" {
  source  = "cloudposse/label/null"
  version = "0.24.1"

  for_each   = local.enabled ? data.aws_vpc_endpoint_service.gateway_endpoint_service : set()
  attributes = [each.key]

  context = module.this.context
}
```

and then

```
resource "aws_vpc_endpoint" "gateway_endpoint" {
  for_each          = local.enabled ? data.aws_vpc_endpoint_service.gateway_endpoint_service : set()
  service_name      = data.aws_vpc_endpoint_service.gateway_endpoint_service[each.key].service_name
  policy            = var.gateway_vpc_endpoints[each.key].policy
  vpc_endpoint_type = data.aws_vpc_endpoint_service.gateway_endpoint_service[each.key].service_type
  vpc_id            = var.vpc_id

  tags = module.gateway_endpoint_label.tags
```

@osterman  @Nuru  what do you think?